src/Umbraco.Infrastructure/Persistence/Repositories/Implement/TagRepository.cs

```diff
@@ -392,13 +392,20 @@ private Sql<ISqlContext> GetTaggedEntitiesSql(TaggableObjectTypes objectType, st
     {
         Sql<ISqlContext> sql = GetTagsSql(culture, true);
 
+        if (objectType == TaggableObjectTypes.Content)
+        {
+            sql = sql
+                .InnerJoin<ContentVersionDto>().On<ContentVersionDto, NodeDto>((node, cv) => node.NodeId == cv.NodeId)
+                .InnerJoin<DocumentVersionDto>().On<DocumentVersionDto, ContentVersionDto>((dv, cv) => cv.Id == dv.Id && dv.Published);
```

Comment:
Adding joins is not very performant, I know that in other areas of the CMS we look at the content path to see if the highest ancestor is the Id of the recycle bin (Id: -20, and for media -21 I believe, there's constants for these Ids). I would recommend trying that for this PR first.

I don't know if it would also make sense to do this for media items by the way, I assume they have the same problem?

Reply:
The is a few issuese here. We have entity for media, content and member.

The first thing is all can be trashed or not, so to exclude the the nodes witch are trashed i add to this line 
https://github.com/umbraco/Umbraco-CMS/pull/15554/files/18823f3d8f7aca32a0a50883f558e64ef027e078#diff-6bfd8066a466dcb406442d00e5991f3f8a798dfa9976124fc128476f681fe88aR408 where i test if it trashed or not.
Therefor i dont need to look into the path or parentid. 

The secound thing is the content can also be pulished and unpublished, and we only want the tags from content witch are pulished. The is no way to tell is a NodeDto is un-/ published without the inner joins.

Therefor i added in the condition `if (objectType == TaggableObjectTypes.Content)` to only make the inner joins if it is an entity for content. I know innerJoins is costly but the is no other way to get the PublishedState